ad-insertion/video-analytics-service/app/modules/PipelineManager.py

```diff
@@ -102,24 +107,44 @@ def get_pipeline_parameters(name, version):
         return params_obj
 
     @staticmethod
-    def create_instance(name, version):
+    def create_instance(name, version, request):
         PipelineManager.logger.info("Creating Instance of Pipeline {name}/{v}".format(name=name, v=version))
         if not PipelineManager.is_pipeline_exists(name, version):
             return None
         pipeline_type = PipelineManager.pipelines[name][str(version)]['type']
         PipelineManager.pipeline_id += 1
-        PipelineManager.pipeline_instances[PipelineManager.pipeline_id] = PipelineManager.pipeline_types[pipeline_type](
-            PipelineManager.pipeline_id,
-            PipelineManager.pipelines[name][str(version)],
-            ModelManager.models)
-        return PipelineManager.pipeline_instances[PipelineManager.pipeline_id]
+        PipelineManager.pipeline_instances[PipelineManager.pipeline_id] = \
+            PipelineManager.pipeline_types[pipeline_type](PipelineManager.pipeline_id,
+                                                            PipelineManager.pipelines[name][str(version)],
+                                                            ModelManager.models,
+                                                            request)
+        PipelineManager.pipeline_queue.append(PipelineManager.pipeline_id)
+        PipelineManager.start()
+        return PipelineManager.pipeline_id
+
+    @staticmethod
+    def start():
+        if (PipelineManager.MAX_RUNNING_PIPELINES <= 0 or PipelineManager.currently_running_pipelines < PipelineManager.MAX_RUNNING_PIPELINES) and len(PipelineManager.pipeline_queue) != 0:
+            pipeline_to_start = PipelineManager.pipeline_instances[PipelineManager.pipeline_queue.popleft()]
+            if(pipeline_to_start is not None):
+                PipelineManager.currently_running_pipelines += 1
+                pipeline_to_start.start()
+        
+    @staticmethod
+    def start_queued():
+        PipelineManager.currently_running_pipelines -= 1
```

Comment:
this seems odd to have this decrement here - maybe a seperate "pipeline_finished" call instead of "start_queued()" - pipeline can call "finished" which can decrement and start if there is a queued one .... So probably just a name change o f this method makes it clearer

Reply:
done